src/diffusers/models/attention.py

```diff
@@ -252,7 +252,7 @@ def __init__(
                 ada_norm_bias,
                 "layer_norm",
             )
-        elif not self.use_ada_layer_norm_single:
+        elif not self.use_ada_layer_norm_single and norm_type != "layer_norm_i2vgen":
```

Comment:
```suggestion
        elif norm_type in ["layer_norm", "ada_norm", "ada_norm_zero"]:
        
```

Can we try to refactor the whole layer norm logic here? Instead of creating booleans, let's write if-else statements using types + string matching

Reply:
Done in be518c87.